File: cpp/test/leetcode/LC0843GuessTheWordTest.cpp
#include <gtest/gtest.h>

#include "leetcode/LC0843GuessTheWord.hpp"

TEST(leetcode, LC0843GuessTheWord_Testcase1) {
    const std::string secret = "acckzz";
    const std::vector<std::string> words = {"acckzz","ccbazz","eiowzz","abcczz"};
    constexpr uint8_t allowed_guesses = 10;
    LC0843GuessTheWord::Master master(secret, words, allowed_guesses);
    LC0843GuessTheWord::findSecretWord(words, master);
    ASSERT_TRUE(master.isGuessed());
}

TEST(leetcode, LC0843GuessTheWord_Testcase2) {
    const std::string secret = "hamada";
    const std::vector<std::string> words = {"hamada","khaled"};
    constexpr uint8_t allowed_guesses = 10;
    LC0843GuessTheWord::Master master(secret, words, allowed_guesses);
    LC0843GuessTheWord::findSecretWord(words, master);
    ASSERT_TRUE(master.isGuessed());
}

File: java/src/test/java/com/bouffalant/leetcode/LC0843GuessTheWordTest.java
package com.bouffalant.leetcode;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class LC0843GuessTheWordTest {
    @Test
    void testcase1() {
        String secret = "acckzz";
        String[] words = {"acckzz","ccbazz","eiowzz","abcczz"};
        int allowedGuesses = 10;
        LC0843GuessTheWord.Master master = new LC0843GuessTheWord.Master(secret, words, allowedGuesses);
        LC0843GuessTheWord.findSecretWord(words, master);
        assertTrue(master.isGuessed());
    }

    @Test
    void testcase2() {
        String secret = "hamada";
        String[] words = {"hamada","khaled"};
        int allowedGuesses = 10;
        LC0843GuessTheWord.Master master = new LC0843GuessTheWord.Master(secret, words, allowedGuesses);
        LC0843GuessTheWord.findSecretWord(words, master);
        assertTrue(master.isGuessed());
    }
}
